libs/sdk-ui-kit/src/Dialog/EmbedInsightDialog/EmbedInsightDialogBase/components/EmbedInsightContent.tsx

```diff
@@ -21,29 +20,25 @@ interface IEmbedInsightContentProps {
  * @internal
  */
 export const EmbedInsightContent: React.FC<IEmbedInsightContentProps> = (props) => {
-    const {
-        integrationDocLink,
-        embedTypeOptions,
-        embedTab,
-        code,
-        openSaveInsightDialog,
-        onCopyCode,
-        onOptionsChange,
-    } = props;
+    const { integrationDocLink, embedTypeOptions, code, openSaveInsightDialog, onCopyCode, onOptionsChange } =
+        props;
+
+    const renderEmbedOptions = useCallback(() => {
+        return embedTypeOptions.type === "react" ? (
+            <ReactOptions option={embedTypeOptions as IReactOptions} onChange={onOptionsChange} />
+        ) : (
+            <WebComponentsOptions
+                option={embedTypeOptions as IWebComponentsOptions}
+                onChange={onOptionsChange}
+            />
+        );
+    }, [embedTypeOptions, onOptionsChange]);
+
     return (
         <div className="embed-insight-dialog-content">
             <PrepareEnvMessage integrationDocLink={integrationDocLink} />
             <div className="embed-insight-dialog-code">
-                <div className="embed-insight-dialog-code-settings">
-                    {embedTab === "react" ? (
```

Comment:
Just a confirmation, do we really need to create a new function `renderEmbedOptions`, which is used only at one place?

→ Can we just simply replace `embedTab` with `embedTypeOptions.type` here?

Reply:
No, I think we need to use `useCallback` function to memorize it. This function won't run in case the props `embedTypeOptions, onOptionsChange` don't change. With the old code, it always runs.

Reply:
How about `useMemo` then? I agree that with `useCallback` the value of `renderEmbedOptions` itself does not change if the list of dependencies doesn't change.

However, in this context, we are calling `renderEmbedOptions()`, aren't we?
→ Can we be sure that the return value of the function also remains unchanged?

Reply:
`useMemo` will return a memoized value and `useCallback` will return a memoized function. In this case, I think `useMemo` is more suitable than `useCallback`. I have updated this. Thanks!